linker/linker.cpp

```diff
@@ -882,7 +890,10 @@ static const ElfW(Sym)* dlsym_linear_lookup(android_namespace_t* ns,
 soinfo* find_containing_library(const void* p) {
   ElfW(Addr) address = reinterpret_cast<ElfW(Addr)>(p);
   for (soinfo* si = solist_get_head(); si != nullptr; si = si->next) {
-    if (address >= si->base && address - si->base < si->size) {
+    if ((address >= si->base && address - si->base < si->size)
+        || si->find_rand_segment_info(address) != nullptr) {
+      // FIXME: this assumes that si->size also covers
```

Comment:
@ahomescu Looks like you wrote this. Do you remember why this would be the case? If `si->size` doesn't cover the PF_RAND_ADDR segments, it should be fine as far as I can tell...